store/sqlstore/retention_policy_store.go

```diff
@@ -0,0 +1,529 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package sqlstore
+
+import (
+	"database/sql"
+	"errors"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/go-sql-driver/mysql"
+	"github.com/lib/pq"
+	"github.com/mattermost/gorp"
+	"github.com/mattermost/mattermost-server/v5/einterfaces"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/store"
+)
+
+type SqlRetentionPolicyStore struct {
+	*SqlStore
+	metrics einterfaces.MetricsInterface
+}
+
+func newSqlRetentionPolicyStore(sqlStore *SqlStore, metrics einterfaces.MetricsInterface) store.RetentionPolicyStore {
+	s := &SqlRetentionPolicyStore{
+		SqlStore: sqlStore,
+		metrics:  metrics,
+	}
+
+	for _, db := range sqlStore.GetAllConns() {
+		table := db.AddTableWithName(model.RetentionPolicy{}, "RetentionPolicies")
+		table.SetKeys(false, "Id")
+		table.ColMap("Id").SetMaxSize(26)
+		table.ColMap("DisplayName").SetMaxSize(64)
+
+		tableC := db.AddTableWithName(model.RetentionPolicyChannel{}, "RetentionPoliciesChannels")
+		tableC.SetKeys(false, "ChannelId")
+		tableC.ColMap("PolicyId").SetMaxSize(26)
+		tableC.ColMap("ChannelId").SetMaxSize(26)
+
+		tableT := db.AddTableWithName(model.RetentionPolicyTeam{}, "RetentionPoliciesTeams")
+		tableT.SetKeys(false, "TeamId")
+		tableT.ColMap("PolicyId").SetMaxSize(26)
+		tableT.ColMap("TeamId").SetMaxSize(26)
+	}
+
+	return s
+}
+
+func (s *SqlRetentionPolicyStore) createIndexesIfNotExists() {
+	s.CreateCompositeIndexIfNotExists("IDX_RetentionPolicies_DisplayName_Id", "RetentionPolicies",
+		[]string{"DisplayName", "Id"})
+	s.CreateIndexIfNotExists("IDX_RetentionPoliciesChannels_PolicyId", "RetentionPoliciesChannels", "PolicyId")
+	s.CreateIndexIfNotExists("IDX_RetentionPoliciesTeams_PolicyId", "RetentionPoliciesTeams", "PolicyId")
+	s.CreateForeignKeyIfNotExists("RetentionPoliciesChannels", "PolicyId", "RetentionPolicies", "Id", true)
+	s.CreateForeignKeyIfNotExists("RetentionPoliciesTeams", "PolicyId", "RetentionPolicies", "Id", true)
+}
+
+// executePossiblyEmptyQuery only executes the query if it is non-empty. This helps avoid
+// having to check for MySQL, which, unlike Postgres, does not allow empty queries.
```

Comment:
Why not just not execute it the query for *both* drivers if it's empty?

Reply:
I'm not sure if I understand your question. The MySQL driver does not allow empty queries.